lib/semantics/resolve-names.cc

```diff
@@ -3554,9 +3556,15 @@ void DeclarationVisitor::CheckSaveStmts() {
     if (auto *symbol{currScope().FindCommonBlock(name)}) {
       auto &objects{symbol->get<CommonBlockDetails>().objects()};
       if (objects.empty()) {
-        Say(name,
-            "'%s' appears as a COMMON block in a SAVE statement but not in"
-            " a COMMON statement"_err_en_US);
+        if (currScope().kind() != Scope::Kind::Block) {
+          Say(name,
+              "'%s' appears as a COMMON block in a SAVE statement but not in"
+              " a COMMON statement"_err_en_US);
+        } else {  // C1108
+          Say(name,
+              "'%s' specifier is not allowed in a BLOCK Construct."
+              " It is a Common Block Name"_err_en_US);
```

Comment:
So far we haven't been using multi-sentence error messages. I suggest wording it like this:
"SAVE statement in BLOCK construct may not contain a common block name"

Reply:
Done